src/lib/containers/widgets/query-filter/QueryFilter.tsx

```diff
@@ -102,6 +102,27 @@ export const applyChangesToValuesColumn = (
   onChangeFn(result)
 }
 
+// This handles multiple checkbox selection with delay refresh
+export const applyMultipleChangesToValuesColumn = (
+  lastRequest: QueryBundleRequest | undefined,
+  facet: FacetColumnResultValues,
+  onChangeFn: Function,
+  faceNameMap?: {}
+) => {
+  const facetNames = faceNameMap && Object.keys(faceNameMap) || []
+  if (facetNames.length) {
+    facet.facetValues.forEach(facetValue => {
+      if (facetNames.includes(facetValue.value)) {
+        facetValue.isSelected = faceNameMap ? faceNameMap[facetValue.value] : false
```

Comment:
I am trying to be careful here. I am checking not just facetNameMap[facetValue.value] is defined but also whether it stores the boolean true or false.

Reply:
ah right, couldn't you change line 115 to the following then -

```tsx
facetValue.isSelected = faceNameMap[facetValue.value] === true
```

Reply:
I tried your code but I got flagged by the linter saying that facetNameMap could be undefined. I guess it's because I have facetNameMap as an optional parameter.